tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -0,0 +1,206 @@
+assert = chai.assert
+
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+class FakeAnchor extends Annotator.Anchor
+
+  _getSegment: -> "html segment for " + @id
+
+  constructor: (manager, annotation, target) ->
+    super manager, annotation, target, 42, 42,
+      "fake quote for" + target.id
+
+    @id = "fake anchor for " + target.id
+
+describe 'Annotator.Plugin.EnhancedAnchoring', ->
+  sandbox = null
+  annotator = null
+  data = null
+  ann1 = null
+  ann2 = null
+  ann3 = null
+  target11 = null
+  target12 = null
+  target2 = null
+  target3 = null
+  anchor11 = null
+  anchor12 = null
+  anchor2 = null
+  anchor3 = null
+
+  setupManager = ->
+    annotator =
+      publish: (event) ->
+
+    am = new Annotator.Plugin.EnhancedAnchoring()
+    am.annotator = annotator
+    am.pluginInit()
+
+    am._chooseAccessPolicy()
+
+    am.strategies.push
+      name: "dummy anchoring strategy"
+      code: (annotation, target) ->
+        new FakeAnchor am,annotation, target
```

Comment:
Oh, never mind, that `FakeAnchor` object *is* how this is supposed to be implemented. To me it looked like we were subclassing it for the sake of testing but this is actually how it's supposed to be implemented.

Could you add comment above the FakeAnchor object to clarify that this is actually a valid implementation of an anchor. Also perhaps we should rename it to `TestAnchor` the "fake" implies that it is in someway a stub or mock.